File: tests/fixtures/check_then_act/ts/unit_of_work_negative_fixture.ts
// The check and the write share one synchronous unit of work, so nothing can land between them.
export class Runtime {
  constructor(private readonly db: any, private readonly store: any) {}

  recordReturn(input: { id: string }) {
    return this.db.unitOfWork(() => {
      const execution = this.store.require(input.id);
      if (execution.status !== 'running') return { status: 'rejected' };
      this.store.append({ id: input.id, kind: 'returned' });
      return { status: 'accepted' };
    });
  }
}

File: tests/fixtures/check_then_act/py/unit_of_work_negative_fixture.py
# The check and the write share one synchronous unit of work, so nothing can land between them.
class Runtime:
    def __init__(self, db, store):
        self.db, self.store = db, store

    def record_return(self, execution_id):
        with self.db.unit_of_work():
            execution = self.store.require(execution_id)
            if execution.status != 'running':
                return 'rejected'
            self.store.append(execution_id, 'returned')
            return 'accepted'
